core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.testgrid.core;
+
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.Assert;
+import org.testng.IObjectFactory;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.testgrid.automation.TestEngineImpl;
+import org.wso2.carbon.testgrid.common.Database;
+import org.wso2.carbon.testgrid.common.Deployment;
+import org.wso2.carbon.testgrid.common.Infrastructure;
+import org.wso2.carbon.testgrid.common.InfrastructureProvider;
+import org.wso2.carbon.testgrid.common.OperatingSystem;
+import org.wso2.carbon.testgrid.common.ProductTestPlan;
+import org.wso2.carbon.testgrid.common.TestPlan;
+import org.wso2.carbon.testgrid.common.TestScenario;
+import org.wso2.carbon.testgrid.common.exception.InfrastructureProviderInitializationException;
+import org.wso2.carbon.testgrid.common.exception.TestAutomationEngineException;
+import org.wso2.carbon.testgrid.common.exception.TestGridConfigurationException;
+import org.wso2.carbon.testgrid.common.exception.TestGridException;
+import org.wso2.carbon.testgrid.common.exception.TestGridInfrastructureException;
+import org.wso2.carbon.testgrid.common.exception.UnsupportedProviderException;
+import org.wso2.carbon.testgrid.deployment.DeployerServiceImpl;
+import org.wso2.carbon.testgrid.infrastructure.InfrastructureProviderFactory;
+import org.wso2.carbon.testgrid.infrastructure.providers.OpenStackProvider;
+import org.wso2.carbon.testgrid.reporting.TestReportEngineImpl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+
+/**
+ * Test class to test the functionality of the {@link TestGridMgtServiceImpl}.
+ *
```

Comment:
Shall we include `@since <version>` since it gives ad idea on from which version this class is included from

Reply:
Fixed with 14598a57750f800c12e6cb5a3828474056e7535c.